### PR TITLE
NAS-117177 / 22.12 / Finish Space Management Card

### PR DESCRIPTION
**Pull request by denysbutenko**
Still need to determine real values. Feedback appreciated.

https://user-images.githubusercontent.com/351613/179421214-c73e1133-b814-4d6c-91f2-392f3c36fe5d.mp4

### CODE REVIEW

File: src/app/pages/datasets/components/dataset-capacity-management-card/dataset-capacity-management-card.component.html [hunk 1]
```diff
@@ -1,42 +1,104 @@
...
+          </span>
+        </ng-container>
+      </div>
+      <div class="details-item">
+        <span class="label">
+          {{ 'Inherited Quotas' | translate }}:
+        </span>
+        <ng-container *ngIf="checkQuotas; else notAvailable">
+          <span class="value" *ngIf="!isLoading; else loaderSkel">
+            {{ inheritedQuotas | filesize: { standard: 'iec', round: 0 } }}
```
Comment (undsoft) [inline src/app/pages/datasets/components/dataset-capacity-management-card/dataset-capacity-management-card.component.html:68]: I think it should be something like 'Inherited Quotas: 50 GiB from my-dataset' and the link on the right should say 'Go to my dataset'

**Review by undsoft — COMMENTED** [review]
Looks pretty good. Some observations:
1. Items with no values (e.g. Reserved 0) should not be shown in the future.
2. Animation on the chart looks nice on the first load, but gets annoying when switching between datasets. I think we should limit it to first run only or remove.
3. When switching between datasets, I will see an old chart until new one is loaded. Best to show a skeleton.

File: src/app/pages/datasets/components/dataset-capacity-management-card/dataset-capacity-management-card.component.ts [hunk 2]
```diff
@@ -32,24 +66,40 @@ export class DatasetCapacityManagementCardComponent implements OnChanges {
...
+
   getQuotas(): void {
-    this.loading = true;
+    this.isLoading = true;
     this.cdr.markForCheck();
     this.quotasSubscription?.unsubscribe();
     this.quotasSubscription = forkJoin([
       this.ws.call('pool.dataset.get_quota', [this.dataset.id, DatasetQuotaType.User, []]),
       this.ws.call('pool.dataset.get_quota', [this.dataset.id, DatasetQuotaType.Group, []]),
+      this.ws.call('pool.dataset.get_quota', [this.dataset.id, DatasetQuotaType.Dataset, []]),
```
Comment (undsoft) [inline src/app/pages/datasets/components/dataset-capacity-management-card/dataset-capacity-management-card.component.ts:94]: Does this actually work? I was under impression that you'd have to traverse the topology tree to find the parent with the quota.

**Comment by denysbutenko** [comment]
> Items with no values (e.g. Reserved 0) should not be shown in the future.

@undsoft did you mean to hide an entire row or just to show `None` instead of `0 B`?

**Comment by undsoft** [comment]
I think we should hide it.

**Comment by denysbutenko** [comment]
All items should be fixed now. Remaining one static thing – `Provisioning Type`.

**Comment by denysbutenko** [comment]
https://user-images.githubusercontent.com/351613/179848296-a9c5fb24-31c8-44d0-95e5-74df32a849fb.mp4

**Review by undsoft — APPROVED** [review]
Looks good.
I would hide Inherited Quotas and Applied Quotas if they are none. I would also change `Manage Dataset` link to `Go To Dataset` or `Go To %short dataset name%`.

Let's hide provisioning type for now and add a middleware ticket to expose this setting.

Also tests for new components.

**Comment by denysbutenko** [comment]
Made the latest improvements and merged master branch.
The unit tests will be added separately. Tickets are:
[NAS-117256 Add unit tests for Space Management Chart](https://ixsystems.atlassian.net/browse/NAS-117256)
[NAS-117255 Add units tests for Dataset Capacity Management Card](https://ixsystems.atlassian.net/browse/NAS-117255)

Regarding `Provisioning Type` I created a ticket [NAS-117254](https://ixsystems.atlassian.net/browse/NAS-117254), but it will be addressed in [NAS-116411](https://ixsystems.atlassian.net/browse/NAS-116411)

**Pull request merged**